.github/workflows/python-package.yml

```diff
@@ -17,11 +17,16 @@ on:
   merge_group:
     types: [checks_requested]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
Why not just make it `true`. It looks like you just want to cancel jobs not in main branch, but this workflow will only run on main branch. Plus, make cancel to be `true` will only cancel jobs in the same group, i.e., the same PR or same push to main.

Reply:
> make cancel to be `true` will only cancel jobs in the same group, i.e., the same PR or same push to main

exactly, bu you do not want to cancel any in you default/main branch

Reply:
Do you mean cancel the runs in your forked repo? I don't think it's needed to trigger runs in the forked main branch. I know the current workflow will trigger it, but it doesn't matter. Please correct me.

Reply:
no, all this CI is running on your side and by group, you define a pipe where workflows are executed sequentially. The second `cancel-in-progress` stated that all such pipelines (except the `main`) execute only the last workflow and cancel all previous ones even they are still running

Reply:
![image](https://user-images.githubusercontent.com/3197038/226099226-e5bfaad8-07c7-4458-860a-4605ae6abe79.png)

I believe the CI is running on FLAML side. 

For the cancel param, in the [doc](https://docs.github.com/en/actions/using-jobs/using-concurrency#overview), it says: 
```
To also cancel any currently running job or workflow in the same concurrency group, specify cancel-in-progress: true
```
As far as I understand, it's exactly what we want with concurrency. Would you mind post some figures to show your point. I'm a little bit confused. Sorry for raising the questions and thank you so much for your patience.

Reply:
> As far as I understand, it's exactly what we want with concurrency. Would you mind post some figures to show your point. I'm a little bit confused. Sorry for raising the questions and thank you so much for your patience.

Yes but in case you have several fast merged PRs yielding in commits to main brach you want let finish CI for each of them not just the last one...

Reply:
> > As far as I understand, it's exactly what we want with concurrency. Would you mind post some figures to show your point. I'm a little bit confused. Sorry for raising the questions and thank you so much for your patience.
> 
> Yes but in case you have several fast merged PRs yielding in commits to main brach you want let finish CI for each of them not just the last one...

So if the group name is `group: github.workflow−{{ github.ref }}-${{ github.head_ref || github.run_id }}`, we can safely set true here. Right?

Reply:
if you want to procced only with the last commit to master/main branch, then yes

Reply:
So let's have a model situation CI that takes one hour, and you have pushed 5 times a commit every 10 minutes (times are just to illustrate the case, not realistic atm)

- `cancel-in-progress: true`
  - for branch or PR, when a new commit is pushed, that running CI is stopped
- `cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}`
  - for PR, when a new commit is pushed, that running CI is stopped
  - for the main branch always waits to finish CI

so for open PR the CI will take at most 1hour as all previous jobs are canceled
but for main, you will have finished CI always check so when you check the commit history, you would always know that a particular commit was valid